pkg/operator/ceph/nfs/nfs.go

```diff
@@ -276,6 +297,12 @@ func validateGanesha(context *clusterd.Context, clusterInfo *cephclient.ClusterI
 	// The existence of the pool provided in n.Spec.RADOS.Pool is necessary otherwise addRADOSConfigFile() will fail
 	_, err := cephclient.GetPoolDetails(context, clusterInfo, n.Spec.RADOS.Pool)
 	if err != nil {
```

Comment:
Before trying to create the pool, can we check if this err is because the pool is not found? Otherwise, in a corner case we will try to create the pool even if it is some other random error when the pool already exists.

Reply:
Is "failed to get pool" specific to the pool not existing, or could it happen for other types of failure?

Reply:
> Is "failed to get pool" specific to the pool not existing, or could it happen for other types of failure?

Not entirely sure, going to try to fix it right now